src/realm/sync/noinst/pending_reset_store.hpp

```diff
@@ -0,0 +1,92 @@
+///////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_NOINST_PENDING_RESET_STORE_HPP
+#define REALM_NOINST_PENDING_RESET_STORE_HPP
+
+#include <realm/status.hpp>
+#include <realm/timestamp.hpp>
+#include <realm/sync/client_base.hpp>
+#include <realm/sync/config.hpp>
+#include <realm/sync/protocol.hpp>
+
+#include <realm/sync/noinst/sync_metadata_schema.hpp>
+
+#include <ostream>
+#include <optional>
+
+namespace realm::sync {
+
+class PendingResetStore;
+
+using PendingResetStoreRef = std::shared_ptr<PendingResetStore>;
+
+struct PendingReset {
+    using Action = sync::ProtocolErrorInfo::Action;
+    Timestamp time;
+    ClientResyncMode mode;
+    Action action = Action::NoAction;
+    std::optional<Status> error;
+};
+
+std::ostream& operator<<(std::ostream& os, const sync::PendingReset& pr);
+bool operator==(const sync::PendingReset& lhs, const sync::PendingReset& rhs);
+
+class PendingResetStore {
```

Comment:
Nice refactoring! The SyncMetadataTable helpers didn't exist yet when this metadata was first added.